dart/lib/vm_service_lib.dart

```diff
@@ -2051,8 +2100,8 @@ class BoundField {
   BoundField();
 
   BoundField._fromJson(Map<String, dynamic> json) {
-    decl = createServiceObject(json['decl']);
-    value = createServiceObject(json['value']);
+    decl = createServiceObject(json['decl'], 'FieldRef');
+    value = createServiceObject(json['value'], 'dynamic');
```

Comment:
Hmm, `dynamic` may not do what we want here.

Perhaps flip the logic a bit, and when creating InstanceRef and Instance objects, always return `null` if a `Null` reference? Or, return `null` if the type wasn't == `InstanceRef` or `Instance`? I'd need to think more about this.

Reply:
In this case, `dynamic` is basically just representing that this field / response can have multiple return types. Maybe it would be more appropriate to take in a list of expected return types here to explicitly check to see whether or not `Instance` or `InstanceRef` is expected for that field or return value.

I don't think we always want to replace `Instance::null` with `null` since it's useful to differentiate between an object in Dart code actually being null vs not being present. It's safe to assume that if the spec doesn't specify `InstanceRef` or `Instance` as valid types for something, we can replace them with `null`. Otherwise, we should still return `Instance::null`.

Reply:
> In this case, dynamic is basically just representing that this field / response can have multiple return types. Maybe it would be more appropriate to take in a list of expected return types here to explicitly check to see whether or not Instance or InstanceRef is expected for that field or return value.

Yup, that’s what we’re using dynamic for; passing a list in here of supported types sounds like a great solution.

Reply:
Gotcha! I've uploaded a new change which checks against the original list of return types rather than the resolved returned type.